core/vmpu/src/mpu_armv7m/vmpu_armv7m.c

```diff
@@ -137,6 +137,17 @@ uint32_t vmpu_sys_mux_handler(uint32_t lr, uint32_t msp)
         halt_info = &info;
     }
 
+    /* If we have a Hard Fault due to the Bus or MemManage Fault escalation we'll try to treat
+     * it as the original fault. */
```

Comment:
The handling of escalated hard faults should also be in its own commit, so we can test it independently of other changes.

What happens if we hard fault a second time as a result of trying to handle the escalated fault? Do we enter a hard fault forever loop? Such a situation seems not so great.

Reply:
Made it a separate commit in this PR.

Still need to think whether we can stuck in a Hard Fault loop.

Reply:
Changed the design to reset active bits, so reverted this change.